src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -1181,6 +1181,8 @@ Namespace Microsoft.CodeAnalysis.Operations
                                                      boundForEachStatement.EnumeratorInfo.NeedToDispose,
                                                      knownToImplementIDisposable:=boundForEachStatement.EnumeratorInfo.NeedToDispose AndAlso
                                                                                   boundForEachStatement.EnumeratorInfo.IsOrInheritsFromOrImplementsIDisposable,
+                                                     disposeMethod:=DirectCast(DirectCast(_semanticModel.Compilation, VisualBasicCompilation).GetSpecialTypeMember(SpecialMember.System_IDisposable__Dispose), MethodSymbol),
```

Comment:
>DirectCast(DirectCast(_semanticModel.Compilation, VisualBasicCompilation).GetSpecialTypeMember(SpecialMember.System_IDisposable__Dispose), MethodSymbol) [](start = 68, length = 152)

I think we should simple pass Nothing here and let flow graph builder figure the symbol out

Reply:
But then we end up in the same situation where CFG has to re-infer decisions that the binding phase already knew. We'll probably never change it for VB so it makes sense that VB says 'this is the method I will use'. C# is then free to extend the disposal rules further without us having to special case it in the CFG builder.